lib/vrl/compiler/src/expression/assignment.rs

```diff
@@ -145,6 +145,23 @@ impl Assignment {
 
         Self { variant }
     }
+
+    /// Get a list of targets for this assignment.
+    ///
+    /// For regular assignments, this contains a single target, for infallible
+    /// assignments, it'll contain both the `ok` and `err` target.
+    pub(crate) fn targets(&self) -> Vec<Target> {
+        let mut targets = vec![];
+
+        match &self.variant {
+            Variant::Single { target, .. } => targets.push(target.clone()),
+            Variant::Infallible { ok, err, .. } => {
+                targets.append(&mut vec![ok.clone(), err.clone()])
```

Comment:
If you pushed twice here you could avoid creating this temporary, unnamed vec.

Reply:
Resolved in [f24dfb7](https://github.com/vectordotdev/vector/pull/12546/commits/f24dfb7427b02d367905ab00817a783de1c2a2bb).